### ISSUE TITLE
Fail typecheck when an enum member's value gets updated

### ISSUE THREAD

**Issue by Achxy** [microsoft/pyright#4247]
**The problem**
An enum member's value cannot be modified during runtime but attempts to modify it is not caught during typecheck
```py
from enum import Enum

class Foo(Enum):
    BAR = 1

Foo.BAR.value = 2 # Fails during runtime but passes typecheck
```

**Proposed solution**
Similar to how typechecking fails when an enum class is subclassed, typechecking should also fail when an attempt is made to modify the value of one of its members.
```py
Foo.BAR.value = 2 # This line fails during typecheck as well
```

**Comment by erictraut** [comment]
The `Enum` class contains many weird, non-standard behaviors that are not possible to reflect in the type system. Type checkers like pyright don't attempt to replicate all of the internals of the `Enum` and `EnumMeta` classes. I'm loath to add more special casing for something as esoteric as this. Since this is a false negative, I think it's fine as is. Mypy also doesn't attempt to detect this case.
